### ISSUE TITLE
Fortify Feature Request: Allow custom conditions for "canFortify"

### ISSUE THREAD

**Issue by BojanOro** [acemod/ACEX#166]
It would be nice to accept a code block so that people can specify additional requirements to be able to fortify; things like presence to a supply vehicle, being in a certain squad, etc.

**Comment by Cuel** [comment]
https://ace3mod.com/wiki/frameworkx/fortify-framework.html#13-adding-custom-deploy-handlers

**Comment by BojanOro** [comment]
While that will accomplish the end goal of not allowing a player to place until conditions are met, it is not intuitive for the self interaction option to show up if the player cannot use it.

The deploy handlers execute while the user is placing an object, my suggestion would execute while the player is opening their self interaction. This way if you see the fortify option you know you can use it, and if you cannot use it you will not see the option.

**Comment by PabstMirror** [comment]
You can restrict to a only certian locations

`acex_fortify_locations pushBack X`
where X is anything that `inArea` would accept - https://community.bistudio.com/wiki/inArea

`[myCar, 50, 50, 0, false]` would be 50 meters within your car

**Comment by BojanOro** [comment]
Yes, but things like checking if a player is slotted in as a member of the logistics unit, or if a team has captured a certain objective, or any other complex set of conditions would be nice to have as well.

**Comment by PabstMirror** [comment]
can set 
`acex_fortify_fortifyAllowed = false/true`

**Comment by BojanOro** [comment]
This would enable/disable for the entire team, wouldnt it? I'm thinking of something that can be set for each player.

**Comment by Cuel** [comment]
The global variable will be set initially for the entire team but can then be toggled on/off for each client (e.g in `initPlayerLocal.sqf`, where you can which squad) 

The rest of the conditions can be covered with deploy handlers or the `acex_fortify_onDeployStart` event.
Optionally you can hide the whole menu when player picks up a weapon with CBA player events https://github.com/CBATeam/CBA_A3/wiki/Player-Events

**Comment by Drofseh** [comment]
I'd still like to see this.
